src/prefect/new_task_engine.py

```diff
@@ -0,0 +1,381 @@
+            )
+        else:
+            dynamic_key = uuid4().hex
+        task_run = await client.create_task_run(
+            task=self.task,
+            name=task_run_name,
+            flow_run_id=(
+                getattr(flow_run_ctx.flow_run, "id", None)
+                if flow_run_ctx and flow_run_ctx.flow_run
+                else None
+            ),
+            dynamic_key=dynamic_key,
+            state=Pending(),
+            task_inputs=task_inputs,
+        )
+        return task_run
+
+    @asynccontextmanager
+    async def enter_run_context(self, client: PrefectClient = None):
+        if client is None:
+            client = await self.get_client()
+
+        self.task_run = await client.read_task_run(self.task_run.id)
+
+        with TaskRunContext(
+            task=self.task,
+            log_prints=self.task.log_prints or False,
+            task_run=self.task_run,
+            parameters=self.parameters,
+            result_factory=await ResultFactory.from_autonomous_task(self.task),
+            client=client,
+        ):
+            self.logger = task_run_logger(task_run=self.task_run, task=self.task)
+            yield
+
+    @asynccontextmanager
+    async def start(self):
+        """
+        - check for a cached state
+        - sets state to running
+        - initialize task run logger
+        - update task run name
+        """
+        async with get_client() as client:
+            self._client = client
+            self._is_started = True
+            try:
+                if not self.task_run:
+                    self.task_run = await self.create_task_run(client)
+
+                yield self
+            except Exception:
+                # regular exceptions are caught and re-raised to the user
+                raise
+            except BaseException as exc:
+                # BaseExceptions are caught and handled as crashes
+                await self.handle_crash(exc)
+                raise
+            finally:
+                self._is_started = False
+                self._client = None
+
+    async def get_client(self):
+        if not self._is_started:
+            raise RuntimeError("Engine has not started.")
+        else:
+            return self._client
+
+    def is_running(self) -> bool:
+        if getattr(self, "task_run", None) is None:
+            return False
+        return getattr(self, "task_run").state.is_running()
+
+    def is_pending(self) -> bool:
+        if getattr(self, "task_run", None) is None:
+            return False  # TODO: handle this differently?
+        return getattr(self, "task_run").state.is_pending()
+
+
+async def run_task(
+    task: Task[P, Coroutine[Any, Any, R]],
+    task_run: Optional[TaskRun] = None,
+    parameters: Optional[Dict[str, Any]] = None,
+    wait_for: Optional[Iterable[PrefectFuture[A, Async]]] = None,
+    return_type: Literal["state", "result"] = "result",
+) -> "R | State | None":
```

Comment:
I think it was me who accidentally introduced the 3.10 syntax 😅 
```suggestion
) -> "Union[R, State, None]":
```